worker.py

```diff
@@ -0,0 +1,140 @@
+from email.mime import text
+import logging
+import pika
+import smtplib
+from sqlalchemy.orm import exc
+
+from mailbbgun import app
+from mailbbgun import models
+
+
+if app.config['DEBUG']:
+    logging.basicConfig(level=logging.DEBUG)
+
+
+_BANNER = """
+  __  __       _ _   ____  ____     _____
+ |  \/  |     (_) | |  _ \|  _ \   / ____|
+ | \  / | __ _ _| | | |_) | |_) | | |  __ _   _ _ __
+ | |\/| |/ _` | | | |  _ <|  _ <  | | |_ | | | | '_ \\
+ | |  | | (_| | | | | |_) | |_) | | |__| | |_| | | | |
+ |_|  |_|\__,_|_|_| |____/|____/   \_____|\__,_|_| |_|
+
+        _  ____________.---------.
+        \`'  __________|_________| o o o
+        /   (_)__]
+       |    |
+      .'   .'
+      |____]
+"""
+_DELIVERY_MODE_PERSISTENT = 2
+_LOG = logging.getLogger(__name__)
+
+
+class MailBBGunWorker():
+
+    def configure_rabbitmq(self):
+        self.connection = pika.BlockingConnection(
+            pika.ConnectionParameters(app.config['RABBITMQ_HOST'])
+        )
+        self.channel = self.connection.channel()
+        self.channel.queue_declare(queue='mailbbgun', durable=True)
+        self.channel.queue_bind(exchange='amq.direct', queue='mailbbgun')
+
+        self.retry_channel = self.connection.channel()
+        self.retry_channel.queue_declare(
+            queue='retry_delay',
+            durable=True,
+            arguments={
+                'x-message-ttl': app.config['RETRY_DELAY_MS'],
+                'x-dead-letter-exchange': 'amq.direct',
+                'x-dead-letter-routing-key': 'mailbbgun'
+            }
+        )
+
+    def process_message(self, ch, method, properties, body):
+        message_id = body.decode('UTF-8')
+        _LOG.info("Processing message id {}".format(message_id))
+        try:
+            message = self._get_message_by_id(message_id)
+            email = text.MIMEText(message.text)
+            email['Subject'] = message.subject
+            email['From'] = app.config['FROM_EMAIL']
+            email['To'] = message.to
+            self._send_email(email)
+            self._update_message_status(message, models.Status.delivered)
+            _LOG.info("Delivered message id {}".format(message_id))
+        except (exc.NoResultFound, exc.MultipleResultsFound):
+            _LOG.exception("Fatal database exception on message id {}".format(
+                message_id
+            ))
+        except smtplib.SMTPException:
+            if message.retries < app.config['MAX_RETRIES']:
+                _LOG.warning(
+                    "Failed to deliver message id {}.  Retrying.".format(
+                        message_id
+                    )
+                )
+                self._increment_message_retries(message)
+                self._schedule_retry(message_id)
```

Comment:
Does it mean that if the worker dies here the message won't ever be delivered? If yes what could we do to fix that?